core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/jupyter/KotlinNotebookPluginUtils.kt

```diff
@@ -79,27 +88,80 @@ public object KotlinNotebookPluginUtils {
         }
 
     /**
-     * Sorts the given data frame by the specified columns.
+     * Sorts a dataframe by multiple columns with specified sorting order for each column.
+     * If a column type is not comparable, sorting by string representation is applied instead.
      *
-     * @param df The data frame to be sorted.
-     * @param columnPaths The paths of the columns to be sorted. Each path is represented as a list of strings.
-     * @param isDesc A list of booleans indicating whether each column should be sorted in descending order.
-     *        The size of this list must be equal to the size of the columnPaths list.
-     * @return The sorted data frame.
+     * @param df The dataframe to be sorted.
+     * @param columnPaths A list of column paths where each path is a list of strings representing the hierarchical path of the column.
+     * @param isDesc A list of boolean values indicating whether each column should be sorted in descending order;
+     *               true for descending, false for ascending. The size of this list should match the size of `columnPaths`.
+     * @return The sorted dataframe.
      */
-    public fun sortByColumns(df: AnyFrame, columnPaths: List<List<String>>, isDesc: List<Boolean>): AnyFrame =
-        df.sortBy {
-            require(columnPaths.all { it.isNotEmpty() })
-            require(columnPaths.size == isDesc.size)
+    public fun sortByColumns(df: AnyFrame, columnPaths: List<List<String>>, isDesc: List<Boolean>): AnyFrame {
+        require(columnPaths.all { it.isNotEmpty() })
+        require(columnPaths.size == isDesc.size)
+
+        val sortKeys = columnPaths.map { path ->
+            ColumnPath(path)
+        }
+
+        val comparator = createComparator(sortKeys, isDesc)
 
-            val sortKeys = columnPaths.map { path ->
-                ColumnPath(path)
+        return df.sortWith(comparator)
+    }
+
+    private fun createComparator(sortKeys: List<ColumnPath>, isDesc: List<Boolean>): Comparator<DataRow<*>> {
+        return Comparator { row1, row2 ->
+            for ((key, desc) in sortKeys.zip(isDesc)) {
+                val comparisonResult = if (row1.df().getColumn(key).type.isComparable()) {
```

Comment:
actually, we already have an `AnyCol.isComparable()` function :) it includes the fix I describe in the other comment

Reply:
Thank you for noticing the problem!